### PR TITLE
Feature/fix-sms-signup

### PR DESCRIPTION
**Pull request by tombelieber**
Feature
1. login ✅
2. sms sign up ✅
3. sms verification ✅
4. email verification ✅

Changes
1. added Swagger
2. added GlobalExceptionFilter to refactor boilerplate error capture code
3. `GetUser` -> `GetCognitoUser` decorator check for jwt, accessToken only, not user form db
4. add GetAccessToken decorator for convenient use

Fix
1. add validation pipe to make class-validator working (aka DTOs)

TODO
1. create new user pool, make phone_number NOT required

### CODE REVIEW

**Comment by sing0629** [comment]
Add the testing flow.
How to get the accessToken? signup? login?

File: src/auth/dto/login.dto.ts [hunk 1]
```diff
@@ -3,7 +3,7 @@ import { IsNotEmpty, IsString } from 'class-validator';
 export class LoginDto {
   @IsNotEmpty()
   @IsString()
-  email: string;
+  username: string;
```
Comment (sing0629) [inline src/auth/dto/login.dto.ts:6]: Allow user to use username or email to login?

File: src/auth/cognito/cognito.service.ts [hunk 2]
```diff
@@ -77,115 +78,79 @@ export class CognitoService {
...
   }
 
-  async signUp(signUpDto: SignUpDto) {
+  async signUp(signUpDto: SignUpDto): Promise<SignUpCommandOutput> {
     const secretHash = this.createSecretHash(signUpDto.username);
     const params: SignUpCommandInput = {
       ClientId: this.clientId,
       Password: signUpDto.password,
       Username: signUpDto.username,
-      ValidationData: [{ Name: 'phone_number', Value: '+85267034350' }],
```
Comment (sing0629) [inline src/auth/cognito/cognito.service.ts:101]: ValidationData allows the program chose verify by email or SMS

Comment (tombelieber) [reply src/auth/cognito/cognito.service.ts:101]: since the current user group doesn't allow empty Phone Number input

**Pull request merged**